### PR TITLE
Bloodtype field

### PR DESCRIPTION
**Pull request by SimNee**
Added the basic class for ```bloodtype```, an attribute of ```Persons```

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Bloodtype.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public Bloodtype(String bloodtype) {
+        requireNonNull(bloodtype);
+        checkArgument(isValidBloodtype(bloodtype), MESSAGE_CONSTRAINTS);
+        value = bloodtype;
+    }
+
+    /**
+     * Returns true if a given string is a valid bloodtype.
+     */
+    public static boolean isValidBloodtype(String test) {
```
Comment (C-likethis123) [inline src/main/java/seedu/address/model/person/Bloodtype.java]: Can your isValidBloodType return false for inputs that are not any specific blood type? Refer to William's PR on the Type.java class for an example.

File: src/main/java/seedu/address/model/person/Bloodtype.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Represents a Person's bloodtype in ORGANice.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBloodtype(String)}
+ */
+public class Bloodtype {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Bloodtype should only have A, B, O or AB."
+                    + "Positive bloodtypes will have a '+' behind, it should not be blank";
```
Comment (WilliamRyank) [inline src/main/java/seedu/address/model/person/Bloodtype.java]: can remove this

File: src/main/java/seedu/address/model/person/Bloodtype.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public static final String VALIDATION_REGEX = "\\d+";
```
Comment (WilliamRyank) [inline src/main/java/seedu/address/model/person/Bloodtype.java]: can try regex to this `^(A|B|AB|O)$`

File: src/main/java/seedu/address/model/person/Bloodtype.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        requireNonNull(bloodtype);
+        checkArgument(isValidBloodtype(bloodtype), MESSAGE_CONSTRAINTS);
+        value = bloodtype;
+    }
+
+    /**
+     * Returns true if a given string is a valid bloodtype.
+     */
+    public static boolean isValidBloodtype(String test) {
+        return test.matches(VALIDATION_REGEX);
```
Comment (WilliamRyank) [inline src/main/java/seedu/address/model/person/Bloodtype.java]: then use `test.toUpperCase().matches(VALIDATION_REGEX)`

File: src/main/java/seedu/address/model/person/Bloodtype.java (hunk 1 above)
Comment (SimNee) [reply src/main/java/seedu/address/model/person/Bloodtype.java]: Done, refer to lastest commit

File: src/main/java/seedu/address/model/person/Bloodtype.java (hunk 4 above)
Comment (SimNee) [reply src/main/java/seedu/address/model/person/Bloodtype.java]: used equals.()

**Pull request closed without merging**